src/scheduler.cc

```diff
@@ -95,26 +98,27 @@ __hidden ncclResult_t mscclSchedulerInit() {
     mscclAlgoDirStr += (mscclUnitTestMode && mscclUnitTestMode()) ? mscclUnitTestAlgoDefaultDir : updateAlgoDirByVMSize(std::string(mscclAlgoDefaultDir));
     mscclAlgoDir = mscclAlgoDirStr.c_str();
     // Get share Directory Paths
-    mscclAlgoShareDirStr = selfLibPath.substr(0, selfLibPath.find("lib"));
+    mscclAlgoShareDirStr = selfLibPath.substr(0, selfLibPath.find_last_of("/\\") + 1);
     mscclAlgoShareDirStr += (mscclUnitTestMode && mscclUnitTestMode()) ? mscclUnitTestAlgoShareDirPath : updateAlgoDirByVMSize(std::string(mscclAlgoShareDirPath));
     mscclAlgoShareDir = mscclAlgoShareDirStr.c_str();
   }
-  fprintf(stdout, "MSCCL: External Scheduler will use %s as algorithm directory and %s as share algorithm directory\n", mscclAlgoDir, mscclAlgoShareDir);
+  fprintf(stdout, "%s: External Scheduler will use %s as algorithm directory and %s as share algorithm directory\n", MSCCL_SCHEDULER_NAME, mscclAlgoDir, mscclAlgoShareDir);
   struct dirent *entry = nullptr;
   DIR *dp = nullptr;
   dp = opendir(mscclAlgoDir);
   if (dp == nullptr) {
-    // Try to find the algorithm directory under share folder based on libnccl.so or librccl.so path
+    // Try to find the algorithm directory under share folder based on libmsccl-scheduler.so path
     dp = opendir(mscclAlgoShareDir);
     if (dp == nullptr) {
+      perror("opendir failed");
```

Comment:
why we need another error print?

Reply:
it's debugging purpose. already removed now.